cmake/dfu_multi_image.cmake

```diff
@@ -18,37 +18,42 @@ find_package(Python3 REQUIRED)
 #   OUTPUT        location of the created package
 #
 function(dfu_multi_image_package TARGET_NAME)
-    cmake_parse_arguments(ARG "" "OUTPUT" "IMAGE_IDS;IMAGE_PATHS;DEPENDS" ${ARGN})
-
-    if (NOT DEFINED ARG_IMAGE_IDS OR NOT ARG_IMAGE_PATHS OR NOT ARG_OUTPUT)
-        message(FATAL_ERROR "All IMAGE_IDS, IMAGE_PATHS and OUTPUT arguments must be specified")
-    endif()
-
-    # Prepare dfu_multi_image_tool.py argument list
-    set(SCRIPT_ARGS "create")
-
-    foreach(image IN ZIP_LISTS ARG_IMAGE_IDS ARG_IMAGE_PATHS)
-        list(APPEND SCRIPT_ARGS "--image" "${image_0}" "${image_1}")
-    endforeach()
-
-    list(APPEND SCRIPT_ARGS ${ARG_OUTPUT})
-
-    # Pass the argument list via file to avoid hitting Windows command-line length limit
-    string(REPLACE ";" "\n" SCRIPT_ARGS "${SCRIPT_ARGS}")
-    file(GENERATE OUTPUT ${ARG_OUTPUT}.args CONTENT ${SCRIPT_ARGS})
-
-    add_custom_command(
-        COMMAND
-        ${Python3_EXECUTABLE}
-        ${ZEPHYR_NRF_MODULE_DIR}/scripts/bootloader/dfu_multi_image_tool.py
-        @${ARG_OUTPUT}.args
-        OUTPUT
-        ${ARG_OUTPUT}
-        DEPENDS
-        ${ARG_DEPENDS}
-    )
-    add_custom_target(${TARGET_NAME} ALL
-        DEPENDS
-        ${ARG_OUTPUT}
-    )
+  cmake_parse_arguments(ARG "" "OUTPUT;ALIGN" "IMAGE_IDS;IMAGE_PATHS;DEPENDS" ${ARGN})
+
+  if(NOT DEFINED ARG_IMAGE_IDS OR NOT ARG_IMAGE_PATHS OR NOT ARG_OUTPUT)
+    message(FATAL_ERROR "All IMAGE_IDS, IMAGE_PATHS and OUTPUT arguments must be specified")
```

Comment:
change to WARNING and allow users to fix their configuration rather than breaking the build (can `return()` early), optionally .c can fail with an assert so it doesn't try to build a non working output

Reply:
Fixed